frontend/javascripts/oxalis/view/right-border-tabs/meshes_view.js

```diff
@@ -277,29 +278,43 @@ class MeshesView extends React.Component<Props, State> {
       }
     };
 
-    const getComputeMeshfileTooltip = node => (
-      <Tooltip
-        title={
-          features().jobsEnabled
-            ? "Compute a mesh file for this dataset so that meshes for segments can be loaded quickly afterwards."
-            : "The computation of mesh files is not supported by this webKnossos instance."
-        }
-      >
-        {node}
-      </Tooltip>
-    );
+    const getComputeMeshFileTooltipInfo = () => {
+      let title = "";
+      let disabled = true;
+      if (!features().jobsEnabled) {
+        title = "Computation jobs are not enabled for this webKnossos instance.";
+      } else if (this.props.hasVolume) {
+        title =
+          "Meshes cannot be computed for volume annotations. Please open this dataset in view mode to compute a mesh file";
+      } else if (this.props.segmentationLayer == null) {
+        title = "There is no segmentation layer for which a mesh file could be computed.";
+      } else {
+        title =
+          "Compute a mesh file for this dataset so that meshes for segments can be loaded quickly afterwards.";
+        disabled = false;
+      }
 
-    const getComputeMeshFileButton = () =>
-      getComputeMeshfileTooltip(
-        <Button
-          size="small"
-          loading={this.state.activeMeshJobId != null}
-          onClick={startComputingMeshfile}
-          disabled={!features().jobsEnabled}
-        >
-          Compute Mesh File
-        </Button>,
+      return {
+        disabled,
+        title,
+      };
+    };
+
+    const getComputeMeshFileButton = () => {
+      const { disabled, title } = getComputeMeshFileTooltipInfo();
```

Comment:
Do theses till tip infos apply to the "Compute Mesh" button as well? I think they do, right? A single mesh also needs a segmentation layer and won't work with volume annotations.

Reply:
Now, the ad-hoc mesh button also has a tooltip. It needed slightly different code (e.g., it doesn't depend on `jobsEnabled`) and wording.

Reply:
@hotzenklotz @philippotto I am a bit confused here, should the ad-hoc isosurface computation really be disabled for volume annotations? There exists the `/volume/:tracingId/isosurface` request in the tracingstore, which is now inaccessible?

Reply:
Oh, good catch. I'll open a PR to fix this.